dataland-batch-manager/build.gradle.kts

```diff
@@ -0,0 +1,99 @@
+// dataland-batch-service
+
+val sonarSources by extra(sourceSets.asMap.values.flatMap { sourceSet -> sourceSet.allSource })
+val jacocoSources by extra(sonarSources)
+val jacocoClasses by extra(
+    sourceSets.asMap.values.flatMap { sourceSet ->
+        sourceSet.output.classesDirs.flatMap {
+            fileTree(it) {
+                exclude("**/openApiClient/**")
+            }.files
+        }
+    },
+)
+val jacocoVersion: String by project
+
+plugins {
+    kotlin("jvm")
+    kotlin("plugin.spring")
+    jacoco
+    id("org.springframework.boot")
+    kotlin("kapt")
+}
+
+java.sourceCompatibility = JavaVersion.VERSION_17
+
+dependencies {
+    implementation(libs.moshi.kotlin)
+    implementation(libs.jackson.dataformat.csv)
+    implementation(libs.jackson.kotlin)
+    implementation(libs.okhttp)
+    implementation(libs.commons.io)
+    testImplementation("org.springframework.boot:spring-boot-starter-test")
+    implementation("org.springframework.boot:spring-boot-starter-security")
+}
+
+tasks.test {
+    useJUnitPlatform()
+
+    extensions.configure(JacocoTaskExtension::class) {
+        setDestinationFile(file("$buildDir/jacoco/jacoco.exec"))
+    }
+}
+jacoco {
+    toolVersion = jacocoVersion
+}
+
+tasks.register("generateBackendClient", org.openapitools.generator.gradle.plugin.tasks.GenerateTask::class) {
+    val backendClientDestinationPackage = "org.dataland.datalandbackend.openApiClient"
+    input = project.file("${project.rootDir}/dataland-backend/backendOpenApi.json").path
+    outputDir.set("$buildDir/clients/backend")
+    packageName.set(backendClientDestinationPackage)
+    modelPackage.set("$backendClientDestinationPackage.model")
+    apiPackage.set("$backendClientDestinationPackage.api")
+    generatorName.set("kotlin")
+
+    additionalProperties.set(
+        mapOf(
+            "removeEnumValuePrefix" to false,
+        ),
+    )
+    configOptions.set(
+        mapOf(
+            "dateLibrary" to "java17",
+            "useTags" to "true",
+        ),
+    )
+}
+
+tasks.register("generateClients") {
+    dependsOn("generateBackendClient")
+}
```

Comment:
Die "Sammeltask" `generateClients` ist hier überflüssig, da man ja nur den backend-client braucht, was sich auch wahrscheinlich nicht ändern wird in Zukunft.

Reply:
Wenn man im Dataland Projekt den task `generateClients` ausführt, würde ich aber erwarten dass überall alle Clients neu generiert werden. Daher würde ich diese Task überall mitziehen.

Reply:
Ja hast recht, das habe ich nicht bedacht. Ich resolve.